implementations/rust/ockam/ockam_identity/src/service.rs

```diff
@@ -0,0 +1,229 @@
+
+        let res: Response = dec.decode()?;
+
+        if let Some(Status::Ok) = res.status() {
+        } else {
+            return Err(IdentityError::ConsistencyError.into());
+        }
+
+        let _res: ImportResponse = dec.decode()?;
+
+        Ok(())
+    }
+
+    async fn verify_and_add_contact(
+        ctx: &mut Context,
+        identity_id: &str,
+        contact: Vec<u8>,
+        service_address: &str,
+    ) -> Result<()> {
+        let body = VerifyAndAddContactRequest::new(contact);
+        let req = Request::post(format!("identities/{}/verify_and_add_contact", identity_id))
+            .body(body)
+            .to_vec()?;
+
+        let receiving_buf: Vec<u8> = ctx.send_and_receive(route![service_address], req).await?;
+        let mut dec = Decoder::new(&receiving_buf);
+
+        let res: Response = dec.decode()?;
+
+        if let Some(Status::Ok) = res.status() {
+        } else {
+            return Err(IdentityError::ConsistencyError.into());
+        }
+
+        Ok(())
+    }
+
+    async fn create_proof(
+        ctx: &mut Context,
+        identity_id: &str,
+        state: &[u8],
+        service_address: &str,
+    ) -> Result<Vec<u8>> {
+        let body = CreateAuthProofRequest::new(state);
+        let req = Request::post(format!("identities/{}/create_auth_proof", identity_id))
+            .body(body)
+            .to_vec()?;
+
+        let receiving_buf: Vec<u8> = ctx.send_and_receive(route![service_address], req).await?;
+        let mut dec = Decoder::new(&receiving_buf);
+
+        let res: Response = dec.decode()?;
+
+        if let Some(Status::Ok) = res.status() {
+        } else {
+            return Err(IdentityError::ConsistencyError.into());
+        }
+
+        let res: CreateAuthProofResponse = dec.decode()?;
+
+        Ok(res.proof().to_vec())
+    }
+
+    async fn verify_proof(
+        ctx: &mut Context,
+        identity_id: &str,
+        peer_identity_id: &str,
+        state: &[u8],
+        proof: &[u8],
+        service_address: &str,
+    ) -> Result<bool> {
+        let body = VerifyAuthProofRequest::new(peer_identity_id, state, proof);
+        let req = Request::post(format!("identities/{}/verify_auth_proof", identity_id))
+            .body(body)
+            .to_vec()?;
+
+        let receiving_buf: Vec<u8> = ctx.send_and_receive(route![service_address], req).await?;
+        let mut dec = Decoder::new(&receiving_buf);
+
+        let res: Response = dec.decode()?;
+
+        if let Some(Status::Ok) = res.status() {
+        } else {
+            return Err(IdentityError::ConsistencyError.into());
+        }
+
+        let res: VerifyAuthProofResponse = dec.decode()?;
+
+        Ok(res.verified())
+    }
+
+    #[ockam_macros::test]
+    async fn full_flow(ctx: &mut Context) -> Result<()> {
+        let vault1 = Vault::create();
+        let vault2 = Vault::create();
+        // Start services
+        IdentityService::create(ctx, "01", vault1.async_try_clone().await?).await?;
```

Comment:
I don't really get why create multiple services here.

Reply:
Just to simulate restarts and make sure everything is still working. It could have been 1 service